src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java

```diff
@@ -428,104 +400,79 @@ public void simulateLoginStatusChange(boolean login) {
       authDataStore.clearStoredOAuthData();
       return;
     }
 
-    accessToken = savedAuthState.getAccessToken();
-    refreshToken = savedAuthState.getRefreshToken();
-    accessTokenExpiryTime = savedAuthState.getAccessTokenExpiryTime();
-    this.email = savedAuthState.getStoredEmail();
-
-    isLoggedIn = true;
-
     if (!oAuthScopes.equals(savedAuthState.getStoredScopes())) {
       loggerFacade.logWarning(
           "OAuth scope set for stored credentials no longer valid, logging out.");
       loggerFacade.logWarning(oAuthScopes + " vs. " + savedAuthState.getStoredScopes());
-      logOut(false);
+      authDataStore.clearStoredOAuthData();
+      return;
     }
 
+    accessToken = savedAuthState.getAccessToken();
+    refreshToken = savedAuthState.getRefreshToken();
+    accessTokenExpiryTime = savedAuthState.getAccessTokenExpiryTime();
+    email = savedAuthState.getStoredEmail();
     oAuth2Credential = makeCredential();
-  }
 
-  private boolean checkLoggedIn(String message) {
-    if (!isLoggedIn) {
-      if (!logIn(message)) {
-        return false;
-      }
-      uiFacade.notifyStatusIndicator();
-    }
-    return true;
+    isLoggedIn = true;
   }
 
   private void notifyLoginStatusChange(boolean login) {
     synchronized(listeners) {
       for (LoginListener listener : listeners) {
         try {
           listener.statusChanged(login);
-        } catch (Throwable t) {
-          loggerFacade.logError("Exception in LoginListener", t);
+        } catch (Throwable thrown) {
+          loggerFacade.logError("Exception in LoginListener", thrown);
         }
       }
     }
   }
 
   private String queryEmail() {
-    String url = GET_EMAIL_URL;
-
-    HttpResponse resp = null;
     try {
-      HttpRequest get = createRequestFactory(null).buildGetRequest(new GenericUrl(url));
-      resp = get.execute();
-      // TODO(nhcohen): Upgrade to Java 7 after moving to separate project:
-      /*try (Scanner scan = new Scanner(resp.getContent())) {
-        String respStr = "";
+      HttpRequest get = createRequestFactory().buildGetRequest(new GenericUrl(GET_EMAIL_URL));
+      HttpResponse resp = get.execute();
+
+      String respStr = "";
+      try (Scanner scan = new Scanner(resp.getContent())) {
         while (scan.hasNext()) {
           respStr += scan.nextLine();
         }
-      }*/
-      Scanner scan = new Scanner(resp.getContent());
-      String respStr = "";
-      while (scan.hasNext()) {
-        respStr += scan.nextLine();
       }
-      scan.close();
 
-      Map<String, String> params = GoogleLoginState.parseUrlParameters(respStr);
-      String userEmail = params.get("email");
+      String userEmail = parseUrlParameters(respStr).get("email");
       if (userEmail == null) {
         throw new Exception("Response from server is invalid.");
       }
-
       return userEmail;
 
-    } catch (Exception e) {
+    } catch (Exception ex) {
```

Comment:
and we shouldn't catch the exception we just threw (if that's what we're doing here.) That is, don't use exceptions for in-method flow control.